File: app/src/main/java/com/electrosim/ElementInterface.java
package com.electrosim;

import android.graphics.Canvas;
import android.view.MotionEvent;

public interface ElementInterface {

	public boolean insideRectCheck(float x,float y);
	
	public void doDraw(Canvas canvas);
	
	public void mouseMove(MotionEvent event);
	
	public void mouseOut(MotionEvent event);
	
	public LinkPointInterface select(MotionEvent event);

	public void deSelect(MotionEvent event);
	public void translate( float deltaX,float deltaY);
	
	public float getX();
	public float getY();
	
	ElementInterface createNew();
}

File: app/src/main/java/com/electrosim/ElementInterface.kt
package com.electrosim

import android.graphics.Canvas
import android.view.MotionEvent

interface ElementInterface {
    fun insideRectCheck(x: Float, y: Float): Boolean
    fun doDraw(canvas: Canvas)
    fun mouseMove(event: MotionEvent)
    fun mouseOut(event: MotionEvent)
    fun select(event: MotionEvent): LinkPointInterface?
    fun deSelect(event: MotionEvent)
    fun translate(deltaX: Float, deltaY: Float)
    fun getX(): Float
    fun getY(): Float
    fun createNew(): ElementInterface
} 